src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs

```diff
@@ -880,31 +887,31 @@ private Task<LockFile> ReadLockFile(string projectLockFilePath)
                 target.TargetFramework = context.RestoreContext.FrameworkName;
                 target.RuntimeIdentifier = context.RestoreContext.RuntimeName;
 
-                foreach (var library in context.Libraries.OrderBy(x => x, new LibraryComparer()))
+                foreach (var match in context.Matches.OrderBy(x => x.Library, new LibraryComparer()))
                 {
-                    if (library.Name == project.Name)
+                    if (match.Library.Name == project.Name)
                     {
                         continue;
                     }
 
-                    var packageInfo = repository.FindPackagesById(library.Name)
-                                                .FirstOrDefault(p => p.Version == library.Version);
+                    var packageInfo = repository.FindPackagesById(match.Library.Name)
+                                                .FirstOrDefault(p => p.Version == match.Library.Version);
 
-                    var projectDependency = projectResolver.FindProject(library.Name);
-                    if (projectDependency != null)
+                    if (match.LibraryType.Equals(LibraryTypes.Project))
                     {
+                        var projectDependency = projectResolver.FindProject(match.Library.Name);
```

Comment:
It would be a unrecoverable internal error if that happened. A library is only marked with `LibraryTypes.Project` if it came from the `projectResolver`